src/reducers/index.js

```diff
@@ -24,26 +25,40 @@ import shallowEqual from 'shallowEqual';
  * Input state set from props
  */
 const DEFAULT_INPUT_STATE = {
-  bufferRowCount: null,
-  columnGroups: [],
+  columnProps: [],
+  columnGroupProps: [],
   elementTemplates: {
     cell: [],
     footer: [],
     groupHeader: [],
     header: [],
   },
-  footerHeight: 0,
-  groupHeaderHeight: 0,
-  headerHeight: 0,
-  height: undefined,
-  maxHeight: 0,
-  rowHeight: 0,
-  rowHeightGetter: () => 0,
-  rowsCount: 0,
-  subRowHeight: 0,
-  subRowHeightGetter: () => 0,
-  width: 0,
-  useGroupHeader: false,
+  elementHeights: {
```

Comment:
I switched grouped these fields into objects to reduce the number of params to selectors & clean things up